modules/xfeatures2d/test/test_surf.cuda.cpp

```diff
@@ -172,7 +172,14 @@ CUDA_TEST_P(CUDA_SURF, Descriptor)
 }
 
 INSTANTIATE_TEST_CASE_P(CUDA_Features2D, CUDA_SURF, testing::Combine(
-    testing::Values(SURF_HessianThreshold(100.0), SURF_HessianThreshold(500.0), SURF_HessianThreshold(1000.0)),
+    testing::Values(
+#if defined (__x86_64__) || defined (_M_X64)
+            SURF_HessianThreshold(100.0), SURF_HessianThreshold(500.0),
```

Comment:
Conditional compilation is not allowed inside macros on MSVS:

```
C:\build\precommit_custom_windows\3.4\opencv_contrib\modules\xfeatures2d\test\test_surf.cuda.cpp(186,1): error C2121: '#': invalid character: possibly the result of a macro expansion [C:\build\precommit_custom_windows\build\modules\xfeatures2d\opencv_test_xfeatures2d.vcxproj]
C:\build\precommit_custom_windows\3.4\opencv_contrib\modules\xfeatures2d\test\test_surf.cuda.cpp(174,1): error C2143: syntax error: missing ')' before 'if' [C:\build\precommit_custom_windows\build\modules\xfeatures2d\opencv_test_xfeatures2d.vcxproj]
C:\build\precommit_custom_windows\3.4\opencv_contrib\modules\xfeatures2d\test\test_surf.cuda.cpp(174,1): error C2672: 'testing::Values': no matching overloaded function found [C:\build\precommit_custom_windows\build\modules\xfeatures2d\opencv_test_xfeatures2d.vcxproj]
```

Reply:
I pushed modification on #2591